common/src/main/java/com/wynntils/features/map/BeaconBeamFeature.java

```diff
@@ -29,55 +32,96 @@ public class BeaconBeamFeature extends Feature {
     @RegisterConfig
     public final Config<CustomColor> waypointBeamColor = new Config<>(CommonColors.RED);
 
+    private static final int RAINBOW_CHANGE_RATE = 10;
+    private CustomColor currentRainbowColor = CommonColors.RED;
+
+    @SubscribeEvent
+    public void onTick(TickEvent event) {
+        int r = currentRainbowColor.r;
+        int g = currentRainbowColor.g;
+        int b = currentRainbowColor.b;
+
+        if (r > 0 && b == 0) {
```

Comment:
I get it that this will change the color, but will it be a rainbow? Why not use a HSV model instead?

Reply:
Looked pretty rainbow-y to me. I have no experience with HSV, so I am not sure how I would implement that.

Reply:
Ok, fair enough.

Reply:
The thing with HSV is that you can take the Hue (the H in the model name) from 0 to 255 (or 0-1.0 or whatever), and it goes through the entire spectrum of hues, from red to violet. In a way that well matches the human perception of a rainbow. And then it is easy to convert to RGB.

Reply:
We use this method in DurabilityArcFeature for example.

Reply:
Perhaps I can fix that when I unhack the rainbow field?

Reply:
Sounds good!